experimental/uefi/channel/src/lib.rs

```diff
@@ -32,25 +33,76 @@ pub mod schema {
 }
 
 /// Length of the entire frame, including the header.
-pub type FrameLength = u32;
+type FrameLength = u32;
 const FRAME_LENGTH_SIZE: usize = 4;
 static_assertions::assert_eq_size!([u8; FRAME_LENGTH_SIZE], FrameLength);
 
-/// In request frames this field contains the id of the method_or_status that is invoked.
-/// In response frames this field contains the status code of the method_or_status invocation.
-pub type Method = u32;
+/// In request frames this field contains the id of the method that is invoked.
+/// In response frames this field contains the status code of the method invocation.
+type MethodOrStatus = u32;
 const METHOD_SIZE: usize = 4;
-static_assertions::assert_eq_size!([u8; METHOD_SIZE], Method);
+static_assertions::assert_eq_size!([u8; METHOD_SIZE], MethodOrStatus);
 
-pub const FRAME_HEADER_SIZE: usize = 8;
+// TODO(#2848): Use this space to send a stream identifier that identifies
+// frames as belonging to the same message. Currently the frames of every message
+// must be sent as an uninterrupted seqeuence. A stream identifier would allow us
+// to interleave messages.
+/// Reserved space to maintain 8 byte alignement.
+const SPACER_SIZE: usize = 4;
+
+#[repr(u32)]
+#[derive(Copy, Clone)]
+enum Flag {
+    /// Atomic frame, a message sent as a single frame
+    AtomicFrame = 0,
+    /// Starts a stream of frames that constitute a message
+    StreamStart = 1,
+    /// Continues a stream of frames that constitute a message
+    StreamContinuance = 2,
+    /// Ends a stream of frames that constitute a message, indicating that the recipient may
+    /// process the message.
+    StreamEnd = 3,
+}
+const FLAG_SIZE: usize = 4;
```

Comment:
4 bytes for a flag seems excessive? I guess this is mostly for alignment?

Reply:
p much, yes. Originally we planned a smaller value for this. But with the oak_idl using u32 for the method_id I had to increase the mehood_id header value, and that point the easiest way to maintain alignment was to make header key 4 bytes long.

Reply:
I still think we should document the format somewhere else, i.e. the reference for the framing format should not be this Rust code, but some markdown file somewhere, which defines the actual offsets and padding etc. within the header, in which case I think flags could be 1 byte, and we could have 3 bytes of padding / alignment (or reserved for future use). Or would we want the spec to say that this is really a 4 bytes flag field?

Reply:
I think we should have the source of truth of the format in this repo though (as a markdown file), since it needs to be kept in sync with the actual implementation, and we need to be able to refer to it when implementing it in other languages (e.g. @ipetr0v will probably have to do that quite soon in the untrusted loader in C++).